File: app/lib/func.dart
import 'dart:ffi';

class Funcionario {
  final int id;
  final String nome;
  final double salario;
  final String dataCadastro;

  const Funcionario({
    required this.id,
    required this.nome,
    required this.salario,
    required this.dataCadastro,
  });

  factory Funcionario.fromJson(Map<String, dynamic> json) {
    return Funcionario(
      id: json['id'],
      nome: json['nome'],
      salario: json['salario'].toDouble(),
      dataCadastro: json['dataCadastro'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'salario': salario,
      'data_cadastro': dataCadastro,
    };
  }
}

File: api/javascript/semBD/func.js
class Funcionario {
    constructor() {
        this._id = 0;
        this._nome = '';
        this._salario = 0;
        this._dataCadastro = '';
    }

    toJSON() {
        return {
            id: this.id,
            nome: this.nome,
            salario: this.salario,
            dataCadastro: this.dataCadastro
        };
    }

    fromJSON(json) {
        this._id = json.id;
        this._nome = json.nome;
        this._salario = json.salario;
        this._dataCadastro = json.dataCadastro;
    }

    get id() {
        return this._id;
    }

    set id(id) {
        this._id = id;
    }

    get nome() {
        return this._nome;
    }

    set nome(nome) {
        this._nome = nome;
    }

    get salario() {
        return this._salario;
    }

    set salario(salario) {
        this._salario = salario;
    }

    get dataCadastro() {
        return this._dataCadastro;
    }

    set dataCadastro(dataCadastro) {
        this._dataCadastro = dataCadastro;
    }
}

module.exports = Funcionario;